### PR TITLE
chore: climate typing

### PR DESCRIPTION
**Pull request by chemelli74**
SSIA

### CODE REVIEW

**Review by wuwentao — APPROVED** [review]

File: custom_components/midea_ac_lan/climate.py [hunk 1]
```diff
@@ -363,107 +380,119 @@ def set_swing_mode(self, swing_mode: str) -> None:
...
+        return cast(
+            float,
+            self._device.get_attribute(C3Attributes.target_temperature)[self._zone],
+        )
 
     @property
-    def current_temperature(self):
-        return None
+    def current_temperature(self) -> float:
+        return 0
```
Comment (Necroneco) [inline custom_components/midea_ac_lan/climate.py:493]: Returning 0 is incorrect, return type should be `-> float | None`

**Review by Necroneco — CHANGES_REQUESTED** [review]

File: custom_components/midea_ac_lan/climate.py (hunk 1 above)
Comment (chemelli74) [reply custom_components/midea_ac_lan/climate.py:493]: If you want None, then we need to change the superclass as well ;-)

File: custom_components/midea_ac_lan/climate.py [hunk 2]
```diff
@@ -363,107 +380,119 @@ def set_swing_mode(self, swing_mode: str) -> None:
...
 
 
 class MideaC3Climate(MideaClimate):
-    _powers = [
-        C3Attributes.zone1_power,
-        C3Attributes.zone2_power,
-    ]
-
-    def __init__(self, device, entity_key, zone):
+    def __init__(self, device: Any, entity_key: str, zone: str) -> None:
```
[the remark this answers is not in the document]
Comment (chemelli74) [reply custom_components/midea_ac_lan/climate.py:420]: I had the doubt but then looking at the rest of the code I went for "str". Let's create a new PR to change it

File: custom_components/midea_ac_lan/climate.py (hunk 1 above)
Comment (Necroneco) [reply custom_components/midea_ac_lan/climate.py:493]: In superclass, it's `float | None`

https://github.com/home-assistant/core/blob/ca89d22a34884b789a4154a3e426f93243041042/homeassistant/components/climate/__init__.py#L603

**Pull request merged**